File: apps/render-ios-playground/render-ios-playground/Domain/Errors/DomainError.swift
import Foundation

/// Domain-specific errors for the render engine
enum DomainError: Error {
    case invalidComponentType(String)
    case invalidComponentId(String)
    case invalidStyleValue(String)
    case componentNotFound(String)
    case invalidScenarioStructure(String)
    case renderingError(String)
    
    var errorDescription: String? {
        switch self {
        case .invalidComponentType(let message):
            return "Invalid component type: \(message)"
        case .invalidComponentId(let message):
            return "Invalid component ID: \(message)"
        case .invalidStyleValue(let message):
            return "Invalid style value: \(message)"
        case .componentNotFound(let message):
            return "Component not found: \(message)"
        case .invalidScenarioStructure(let message):
            return "Invalid schema structure: \(message)"
        case .renderingError(let message):
            return "Rendering error: \(message)"
        }
    }
}

File: apps/render-android-playground/app/src/main/java/com/max_mrtnv/render_android_playground/domain/errors/DomainError.kt
package com.max_mrtnv.render_android_playground.domain.errors

/**
 * Domain-specific errors for the render engine
 */
sealed class DomainError(message: String) : Exception(message) {
    class InvalidComponentType(message: String) : DomainError("Invalid component type: $message")
    class InvalidComponentId(message: String) : DomainError("Invalid component ID: $message")
    class InvalidPropertyValue(message: String) : DomainError("Invalid property value: $message")
    class InvalidFrame(message: String) : DomainError("Invalid frame: $message")
    class ComponentNotFound(message: String) : DomainError("Component not found: $message")
    class InvalidSchemaStructure(message: String) : DomainError("Invalid schema structure: $message")
    class RenderingError(message: String) : DomainError("Rendering error: $message")
}
